lessons/git/git-collaboration-2in1/index.md

```diff
@@ -0,0 +1,291 @@
+# Spolupráce a Git
+
+„Opravdové” programy zřídka vznikají prací jednoho člověka. Víc hlav víc ví, a tak je dobré si na projekt vytvořit tým.
+Každý člen týmu potřebuje mít přístup k práci ostatních. Jak ale zařídit to, aby několik lidí mohlo najednou upravovat
+stejné části (soubory) programu? 
+
+Určitě jste někdy použili nějaký nástroj na sdílení souborů jako je Dropbox nebo OneDrive. 
+Tyto nástroje se většinou moc neumí vyrovnat s tím, když se změní jeden soubor najednou na dvou počítačích.
+Většinou vytvoří 2 konfliktní kopie a je na uživatelích, aby našli v souborech změny a spojili je.
+
+Takový způsob práce není ve velkém projektu s mnoha programátory udržitelný. 
+Kód je většinou v mnoha souborech a přidání jedné nové funkcionality může vyžadovat změnu ve více souborech. 
+Jen si představte jakou paseku by nadělalo pět programátorů v jedné Dropbox složce.
+
+Zdrojový kód se dá poměrně dobře sdílet a přepoužívat, pokud je napsaný podle určitých pravidel. 
+Spolupráce a sdílení kódu je spjato s programováním od nepaměti, už od prvních počítačů.
+Tohle přepoužívání se postupně vyvinulo do fenoménu *open-source* a *svobodného softwaru*.
+
+Dnes si vyzkoušíme, jak spolupracovat s mnoha programátory na jednom projektu pomocí verzovacího nástroje *Git* a služby *GitHub*.
+
+## Open-source a Free software
```

Comment:
Trochu jsem  rozšířil tu teorii o open-source, minulý běh jsem o tom mluvil a byla na to dobrá zpětná vazba.